Tryouts/Plugins/ApplicationPlugins/MorganStanley.ComposeUI.ProcessExplorer/dotnet/src/LocalCollector/ProcessInfoCollector.cs

```diff
@@ -157,103 +141,81 @@ public async Task AddConnectionMonitor(IConnectionMonitor connections)
 
     public async Task AddEnvironmentVariables(EnvironmentMonitorInfo environmentVariables)
     {
-        if (ProcessInformation.EnvironmentVariables.IsEmpty)
-        {
-            lock (_locker)
-            {
-                ProcessInformation.EnvironmentVariables = environmentVariables.EnvironmentVariables;
-            }
-        }
-        else
+        IEnumerable<KeyValuePair<RuntimeInformation, IEnumerable<KeyValuePair<string, string>>>> info;
+
+        lock (_locker)
         {
             foreach (var env in environmentVariables.EnvironmentVariables)
             {
-                lock (_locker)
-                {
-                    ProcessInformation.EnvironmentVariables.AddOrUpdate(env.Key, env.Value, (_, _) => env.Value);
-                }
+                _processInformation.EnvironmentVariables.AddOrUpdate(env.Key, env.Value, (_, _) => env.Value);
             }
 
-        }
-
-        if (_communicator != null)
-        {
-            var info = new List<KeyValuePair<RuntimeInformation, IEnumerable<KeyValuePair<string, string>>>>()
+            info = new List<KeyValuePair<RuntimeInformation, IEnumerable<KeyValuePair<string, string>>>>()
             {
                 new(_runtimeId, environmentVariables.EnvironmentVariables)
             };
+        }
 
+        if (info.Any())
```

Comment:
Again, I don't think it's necessary to change, just a small observation that info is never empty here - it's change-proof to still check it though.